### PR TITLE
TRT-892: Match jobnames, not variants

### PR DESCRIPTION
**Pull request by neisw**
Review changes switching from variant matches to jobname matches

### CODE REVIEW

File: pkg/api/job_runs.go [hunk 1]
```diff
@@ -116,6 +118,78 @@ func FetchJobRun(dbc *db.DB, jobRunID int64, logger *log.Entry) (*models.ProwJob
 	return jobRun, jobRunTestCount, nil
 }
 
+func findReleaseMatchJobNames(dbc *db.DB, jobRun *models.ProwJobRun, compareRelease string, logger *log.Entry) ([]string, int, error) {
```
Comment (dgoodwin) [inline pkg/api/job_runs.go:121]: I think I can see what this is doing but I am not immediately clear why, could you add godoc or PR description to outline the change being made and the algorithm to do so, and maybe provide some small real world examples. I'm unclear about the segment matching as well as how variants continue to play into it.

File: pkg/api/job_runs.go [hunk 2]
```diff
@@ -116,6 +121,99 @@ func FetchJobRun(dbc *db.DB, jobRunID int64, logger *log.Entry) (*models.ProwJob
...
+
+							jobIds, err := query.ProwJobRunIds(dbc, job.ID)
+
+							if err != nil {
+								logger.WithError(err).Errorf("Failed to query job run ids for %d", job.ID)
+								continue
+							}
+
+							totalJobRunsCount += len(jobIds)
+							allJobNames = append(allJobNames, "'"+job.Name+"'")
```
Comment (dgoodwin) [inline pkg/api/job_runs.go:193]: We should log when we find a matching job name, perhaps at debug level, either here or when we get the final list back out of the func.

File: pkg/api/job_runs.go [hunk 3]
```diff
@@ -272,6 +436,13 @@ func runJobRunAnalysis(jobRun *models.ProwJobRun, compareRelease string, jobRunT
 		response.OverallRisk.Reasons = append(response.OverallRisk.Reasons,
 			fmt.Sprintf("%d tests failed in this run: High", len(jobRun.Tests)))
 		return response, nil
+
+	// Return early is this is for a never-stable job
+	case neverStableJob:
+		response.OverallRisk.Level = apitype.FailureRiskLevelLow
+		response.OverallRisk.Reasons = append(response.OverallRisk.Reasons,
+			"Job is marked as never-stable")
+		return response, nil
```
Comment (dgoodwin) [inline pkg/api/job_runs.go:445]: Is it feasible to proceed with test analysis on these so we could be watching risk of jobs we're hoping to get OUT of never-stable, or is that largely impossible? With variant matching probably not feasible, but with job name we might get something useful?

File: pkg/api/job_runs.go [hunk 4]
```diff
@@ -336,6 +527,64 @@ func runJobRunAnalysis(jobRun *models.ProwJobRun, compareRelease string, jobRunT
...
+	}
+
+	// we had the case where etcd-scaling was reporting
+	// high risk using the variant methodology due
+	// to the analysis including more stable non scaling job results
+	// so, we don't want to just return the highest risk analysis
+	// bias is towards jobname analysis at this point
+	// we could check for the number of runs but
+	// that introduces flakiness on which analysis we are using
+	// for now we have a deterministic result based on the jobnames analsysis so return it
```
Comment (dgoodwin) [inline pkg/api/job_runs.go:575]: nit: typo in analysis

File: pkg/api/job_runs_test.go [hunk 5]
```diff
@@ -267,3 +367,78 @@ func TestSubSliceEqual(t *testing.T) {
...
+		},
+		{
+			name:              "variants risk level",
+			variantsRiskLevel: &apitype.Test{CurrentPassPercentage: 100, CurrentRuns: 100, Variants: []string{"aws", "ovn"}},
+			expectedResponse:  []string{"This test has passed 100.00% of 100 runs on release 4.14 [aws ovn] in the last week."},
+			expectedRiskLevel: apitype.FailureRiskLevelHigh.Level,
+			compareRelease:    "4.14",
+			jobNames:          []string{"jobname-4.14", "jobname-4.13"},
+		},
+	}
```
Comment (dgoodwin) [inline pkg/api/job_runs_test.go:434]: Nice tests!

**Review by dgoodwin — COMMENTED** [review]
This looks pretty good to me just a few nits, it seems we'll be able to easily tell which version got used so that should do the tricky nicely.

File: pkg/api/job_runs.go (hunk 2 above)
Comment (dgoodwin) [reply pkg/api/job_runs.go:193]: Disregard I saw later it was in the result message.

File: pkg/api/job_runs.go (hunk 3 above)
Comment (neisw) [reply pkg/api/job_runs.go]: We can try.  If it looks to be at all flaky I say dump it and get value from the other 99% though.

**Pull request merged**